src/infrastructure/selectors/XmlSelectorAdapter.js

```diff
@@ -0,0 +1,41 @@
+const xpath = require('xpath');
+const { XMLSerializer, DOMParser } = require('xmldom');
+const DataSelectorPort = require('../../domain/common/port/DataSelectorPort');
+
+class XmlSelectorAdapter extends DataSelectorPort {
+  constructor() {
+    super();
+    this.domParser = new DOMParser();
+    this.xmlSerializer = new XMLSerializer();
+  }
+
+  canHandle(contentType) {
+    return ['xml', 'html'].includes(contentType);
+  }
+
+  select(data, path) {
+    const doc = this.domParser.parseFromString(data);
+    this.xmlSerializer.serializeToString(doc); // this throw an error when doc is invalid xml
+
+    const expression = path === null ? '//*' : path;
+    const selected = xpath.select(expression, doc);
+
+    if (selected instanceof Array && selected.length === 0) {
```

Comment:
as I remember this is recommended check to arr `Array.isArray([]);`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray